### PR TITLE
add GNU and Rust triplets to Architecture type

### PR DESCRIPTION
**Pull request by kaniini**
This is needed for Melange's `${{host.triplet.gnu}}` and `${{host.triplet.rust}}` support.

### CODE REVIEW

File: pkg/build/types/types.go [hunk 1]
```diff
@@ -83,6 +84,7 @@ type ImageConfiguration struct {
 }
 
 // Architecture represents a CPU architecture for the container image.
+// TODO(kaniini): Maybe this should be its own package at this point?
```
Comment (luhring) [inline pkg/build/types/types.go:87]: Not a bad idea :)

File: pkg/build/types/types.go [hunk 2]
```diff
@@ -171,6 +173,48 @@ func (a Architecture) ToQEmu() string {
...
+	case ppc64le:
+		return fmt.Sprintf("powerpc64le-unknown-linux-%s", suffix)
+	case s390x:
+		return fmt.Sprintf("s390x-ibm-linux-%s", suffix)
+	default:
+		return fmt.Sprintf("%s-unknown-linux-%s", a.ToQEmu(), suffix)
+	}
+}
+
+func (a Architecture) ToRustTriplet(suffix string) string {
```
Comment (luhring) [inline pkg/build/types/types.go:197]: nit: it'd be nice to have a doc comment here for what this is for

**Review by luhring — APPROVED** [review]
LGTM

File: pkg/build/types/types.go [hunk 3]
```diff
@@ -171,6 +173,48 @@ func (a Architecture) ToQEmu() string {
 	}
 }
 
+func (a Architecture) ToTriplet(suffix string) string {
+	switch a {
+	case _386:
+		return fmt.Sprintf("i486-pc-linux-%s", suffix)
```
Comment (amouat) [inline pkg/build/types/types.go:179]: Should this be 686? I know 486 was a thing, but a very long time ago...

**Pull request merged**